### PR TITLE
Added logic to skip the .z.pd function

### PR DESCRIPTION
**Pull request by David2464**
- Added the default .finspace.dataview in the finspace.q file

- Added the .finspace.dataview variable to the notifyhdb function, this can be overwritten in FinTorQ/appconfig.

- Changed reload hdb mode form NO_RESTART to ROLLING.

- Added in the required finspace flags inside the wdb eod.

- Added logic to skip over the .z.pd functions in the wdb.q file as this function will produce an error in finspace. This should be a temporary fix until AWS fix this issue.

- Added in the .wdb.hdbmessage and .wdb.notifyhdb functions to wdb.q file.

### CODE REVIEW

**Review by zanbuchanan — APPROVED** [review]

File: code/processes/wdb.q [hunk 1]
```diff
@@ -160,13 +161,29 @@ endofday:{[pt;processdata]
...
+		$[sortenabled;endofdaysort;informsortandreload] . (savedir;pt;tablist;writedownmode;mergelimits;hdbsettings;mergemethod)
+		if[.finspace.enabled;
+                        changeset:.finspace.createchangeset[.finspace.database];
+                        ];
+		];
+	
 	.lg.o[`eod;"deleting data from ",$[r:writedownmode~`partbyattr;"partsizes";"tabsizes"]];
 	$[r;@[`.merge;`partsizes;0#];@[`.wdb;`tabsizes;0#]];
+	/-notify all hdbs
+	hdbs:distinct raze {exec w from .servers.getservers[x;y;()!();1b;0b]}'[`proctype`procname;(hdbtypes;hdbnames)];
```
Comment (jonathonmcmurray) [inline code/processes/wdb.q:182]: why do we need to do this by name & type? surely we should be able to use just type?

File: code/processes/wdb.q [hunk 2]
```diff
@@ -160,13 +161,29 @@ endofday:{[pt;processdata]
...
+        if[.finspace.enabled;
+                        .lg.o[`createchangeset;"downloading sym file to scratch directory for ",.finspace.database];
+                        .aws.get_latest_sym_file[.finspace.database;getenv[`KDBSCRATCH]];
+			];
 	/ - if save mode is enabled then flush all data to disk
 	if[saveenabled;
 		endofdaysave[savedir;pt];
 		/ - if sort mode enable call endofdaysort within the process,else inform the sort and reload process to do it
-		$[sortenabled;endofdaysort;informsortandreload] . (savedir;pt;tablist;writedownmode;mergelimits;hdbsettings;mergemethod)];
+		$[sortenabled;endofdaysort;informsortandreload] . (savedir;pt;tablist;writedownmode;mergelimits;hdbsettings;mergemethod)
```
Comment (jonathonmcmurray) [inline code/processes/wdb.q:173]:
```suggestion
		$[sortenabled;endofdaysort;informsortandreload] . (savedir;pt;tablist;writedownmode;mergelimits;hdbsettings;mergemethod);
```

File: code/processes/wdb.q [hunk 3]
```diff
@@ -160,13 +161,29 @@ endofday:{[pt;processdata]
...
+		if[.finspace.enabled;
+                        changeset:.finspace.createchangeset[.finspace.database];
+                        ];
```
Comment (jonathonmcmurray) [inline code/processes/wdb.q:176]:
```suggestion
		if[.finspace.enabled;changeset:.finspace.createchangeset[.finspace.database]];
```

File: code/processes/wdb.q [hunk 4]
```diff
@@ -160,13 +161,29 @@ endofday:{[pt;processdata]
...
+		];
+	
 	.lg.o[`eod;"deleting data from ",$[r:writedownmode~`partbyattr;"partsizes";"tabsizes"]];
 	$[r;@[`.merge;`partsizes;0#];@[`.wdb;`tabsizes;0#]];
+	/-notify all hdbs
+	hdbs:distinct raze {exec w from .servers.getservers[x;y;()!();1b;0b]}'[`proctype`procname;(hdbtypes;hdbnames)];
+	$[.finspace.enabled;
+                .finspace.notifyhdb[;changeset] each .finspace.hdbclusters;
+                notifyhdb[;pt] each hdbs
```
Comment (jonathonmcmurray) [inline code/processes/wdb.q:185]: This is added code - how does the HDB get notified currently? Are we changing how that works for on-prem TorQ?

Comment (David2464) [reply code/processes/wdb.q:183]: Ah I see. Currently the hdbreload is called in the endofdaysave function. The addition of notifyhdb here is unnecessary

File: code/processes/wdb.q (hunk 1 above)
Comment (David2464) [reply code/processes/wdb.q:180]: I had just copied this line from the RDB eod. I'm not sure why hdbnames is included but i will look into it now to see if i can find a reason.

File: code/processes/wdb.q [hunk 5]
```diff
@@ -160,13 +160,23 @@ endofday:{[pt;processdata]
...
 		endofdaysave[savedir;pt];
 		/ - if sort mode enable call endofdaysort within the process,else inform the sort and reload process to do it
-		$[sortenabled;endofdaysort;informsortandreload] . (savedir;pt;tablist;writedownmode;mergelimits;hdbsettings;mergemethod)];
+		$[sortenabled;endofdaysort;informsortandreload] . (savedir;pt;tablist;writedownmode;mergelimits;hdbsettings;mergemethod);
+		if[.finspace.enabled;changeset:.finspace.createchangeset[.finspace.database]];
+		];
 	.lg.o[`eod;"deleting data from ",$[r:writedownmode~`partbyattr;"partsizes";"tabsizes"]];
 	$[r;@[`.merge;`partsizes;0#];@[`.wdb;`tabsizes;0#]];
+	/-notify all hdbs
+	hdbs:distinct raze {exec w from .servers.getservers[x;y;()!();1b;0b]}'[`proctype;hdbtypes];
```
Comment (jonathonmcmurray) [inline code/processes/wdb.q:178]: don't think this line is needed anymore

Comment (David2464) [reply code/processes/wdb.q:178]: Oh yes of course

**Review by jonathonmcmurray — APPROVED** [review]

**Pull request merged**